xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-docker/src/main/java/org/xwiki/test/docker/junit5/servletEngine/ServletContainerExecutor.java

```diff
@@ -170,6 +173,77 @@ private void startContainer() throws Exception
         start(this.servletContainer, this.testConfiguration);
     }
 
+    private void throwExceptionForServletEngineNotYetSupported(TestConfiguration testConfiguration)
+    {
+        throw new RuntimeException(String.format("Servlet engine [%s] is not yet supported!",
+            testConfiguration.getServletEngine()));
+    }
+
+    private GenericContainer getServletContainer(TestConfiguration testConfiguration)
+    {
+        final String baseImageName;
+
+        final GenericContainer container;
+
+        switch (testConfiguration.getServletEngine()) {
+            case TOMCAT:
+                baseImageName = String.format("tomcat:%s", testConfiguration.getServletEngineTag() != null
+                    ? testConfiguration.getServletEngineTag() : LATEST);
+                break;
+
+            case JETTY:
+                baseImageName = String.format("jetty:%s", testConfiguration.getServletEngineTag() != null
+                    ? testConfiguration.getServletEngineTag() : LATEST);
+                break;
+
+            case WILDFLY:
+                baseImageName = String.format("jboss/wildfly:%s", testConfiguration.getServletEngineTag() != null
+                    ? testConfiguration.getServletEngineTag() : LATEST);
+                break;
+
+            default:
+                this.throwExceptionForServletEngineNotYetSupported(testConfiguration);
+                baseImageName = null;
+        }
+
+        // we want to use libreoffice so we need a custom image
+        if (testConfiguration.isOffice()) {
+            // name of the image that we will create
+            String imageName = StringUtils.join(Arrays.asList("xwiki", baseImageName, "office"),
+                CUSTOM_IMAGE_NAME_SEPARATOR).replace(':', CUSTOM_IMAGE_NAME_SEPARATOR);
+
+            // we won't delete the image, so it's possible that the image already exists: it would avoid us to create
+            // it again
```

Comment:
You should always explain why not what. Why would you want to delete the image locally?